streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java

```diff
@@ -490,6 +507,101 @@ private String createYamlFile (TopologyLayout topology) throws Exception {
+                configurationForCluster.put("principal", (String) sc.get("principal"));
+                configurationForCluster.put("keytabPath", (String) sc.get("keytabPath"));
+                clusterToConfiguration.put(clusterName, configurationForCluster);
+            });
+        }
+
+        if (clusterToConfiguration.isEmpty()) {
+            // it will function only when user input keytab path and principal
+            return;
+        }
+
+        putServiceSpecificCredentialConfig(topologyConfig, topologyDag, clusterToConfiguration,
+                Collections.singletonList(HdfsSource.class),
+                Collections.singletonList(HdfsSink.class),
+                Constants.HDFS.SERVICE_NAME, "hdfs_", "hdfs.keytab.file", "hdfs.kerberos.principal",
+                "hdfsCredentialsConfigKeys", "org.apache.storm.hdfs.common.security.AutoHDFS");
+
+        putServiceSpecificCredentialConfig(topologyConfig, topologyDag, clusterToConfiguration,
+                Collections.emptyList(),
+                Collections.singletonList(HBaseSink.class),
+                Constants.HBase.SERVICE_NAME, "hbase_", "hbase.keytab.file", "hbase.kerberos.principal",
+                "hbaseCredentialsConfigKeys", "org.apache.storm.hbase.security.AutoHBase");
+    }
+
+    private void putServiceSpecificCredentialConfig(Config topologyConfig, TopologyDag topologyDag,
+                                                    Map<String, Map<String, String>> clusterToConfiguration,
+                                                    List<Class<?>> outputComponentClasses,
+                                                    List<Class<?>> inputComponentClasses,
+                                                    String serviceName, String clusterKeyPrefix,
+                                                    String keytabPathKeyName, String principalKeyName,
+                                                    String credentialConfigKeyName,
+                                                    String topologyAutoCredentialClassName) {
+
+        boolean componentExists = false;
+        for (OutputComponent outputComponent : topologyDag.getOutputComponents()) {
+            for (Class<?> clazz : outputComponentClasses) {
+                if (outputComponent.getClass().isAssignableFrom(clazz)) {
+                    componentExists = true;
+                    break;
+                }
+            }
+        }
+
+        if (!componentExists) {
+            for (InputComponent inputComponent : topologyDag.getInputComponents()) {
+                for (Class<?> clazz : inputComponentClasses) {
+                    if (inputComponent.getClass().isAssignableFrom(clazz)) {
+                        componentExists = true;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (componentExists) {
+            List<String> clusterKeys = new ArrayList<>();
+
+            clusterToConfiguration.keySet()
```

Comment:
We are using Streamline topology security config, so if users don't set anything for that, we don't add anything to Storm topology config.